### PR TITLE
Feature/zen 14717

### PR DESCRIPTION
**Pull request by burbel**

### CODE REVIEW

File: src/zenoss/toolbox/zencatalogscan.py [hunk 1]
```diff
@@ -126,16 +126,61 @@ def scan_catalog(catalog_name, catalog_list, fix, max_cycles, dmd, log):
...
+                    number_of_issues += 1
+                    broken_rids.append(rid)
+
+            log.info("Paths/UIDs check found %s issues - attempting to remove", len(broken_rids))
+
+            for item in broken_rids:
+              try:
+                catalog_reference.paths.pop(item)
+                catalog_reference.data.pop(item)
+              except:
```
Comment (jhanson) [inline src/zenoss/toolbox/zencatalogscan.py]: Separate out this so if it is in data  but not in paths we can still remove it

File: src/zenoss/toolbox/zencatalogscan.py [hunk 2]
```diff
@@ -126,16 +126,61 @@ def scan_catalog(catalog_name, catalog_list, fix, max_cycles, dmd, log):
...
-    print("[%s] Examining %-35s (%d Objects)" %
-          (time.strftime("%Y-%m-%d %H:%M:%S"), catalog_name, initial_catalog_size))
-    log.info("Examining %s catalog with %d objects" % (catalog_name, initial_catalog_size))
-
     number_of_issues = -1
     current_cycle = 0
     if not fix:
         max_cycles = 1
 
+    # Fix for ZEN-14717 (only for global_catalog, and only if fix==True)
```
Comment (jhanson) [inline src/zenoss/toolbox/zencatalogscan.py]: Extract this chunk about the rid clean up into a separate method

File: src/zenoss/toolbox/zencatalogscan.py [hunk 3]
```diff
@@ -126,16 +126,61 @@ def scan_catalog(catalog_name, catalog_list, fix, max_cycles, dmd, log):
...
+                    chunk_number = scanned_count // progress_bar_chunk_size
+                    scan_progress_message(False, fix, current_cycle, catalog_name, number_of_issues, chunk_number, log)
+                if path not in catalog_reference.uids:
+                    number_of_issues += 1
+                    broken_rids.append(rid)
+
+            log.info("Paths/UIDs check found %s issues - attempting to remove", len(broken_rids))
+
+            for item in broken_rids:
+              try:
```
Comment (jhanson) [inline src/zenoss/toolbox/zencatalogscan.py]: make sure all the indentation levels are consistent

**Comment by burbel** [comment]
Integrated feedback from Joseph.  Here's an updated output when run via command line:

[2015-07-01 10:04:37] Examining global_catalog RIDs consistency     (673098 Objects)
[2015-07-01 10:04:47]  Clean # 1 [==================================================] 100% [5424 Issues Detected]
[2015-07-01 10:05:05]  Verified  [==================================================] 100% 
[2015-07-01 10:05:05] Examining global_catalog                      (673098 Objects)
[2015-07-01 10:07:01]  Scanning  [=========                                         ]  18%

File: src/zenoss/toolbox/zencatalogscan.py (hunk 2 above)
Comment (burbel) [inline src/zenoss/toolbox/zencatalogscan.py]: Moved to separate function, logic and flow cleaned to better match the normal scan process.

File: src/zenoss/toolbox/zencatalogscan.py (hunk 1 above)
Comment (burbel) [inline src/zenoss/toolbox/zencatalogscan.py]: Attempted to also do the scan for data.iteritems() against uids, but every item in data.iteritems() was not in uids so skipping that case for now (was not reported by either customer).

**Pull request merged**